backend/src/services/authDiscordService.ts

```diff
@@ -0,0 +1,41 @@
+import { generateNewTokenPair } from "@services/authService";
+import { discordOauth2Client } from "@/clients/discord";
+import { prisma } from "@/database/db";
+
+const scopes = ["identify", "email"];
+const randomStateValue = Math.random().toString(16).substr(2);
+
+export const regDiscordLink = async () => {
+  const client = discordOauth2Client();
+  const url = client.generateAuthUrl({
+    scope: scopes,
+    state: randomStateValue,
```

Comment:
It will work only if we have one instance of our application and for every user this state will be the same, I think we should generate this `randomStateValue` during request and connect it to user somehow. Lets create separate task for it and remove this `state` here for now